packages/react/src/wallet/adapters/LogoutPresenter.ts

```diff
@@ -1,11 +1,20 @@
 import { ILogoutPresenter, LogoutData } from '@lens-protocol/domain/use-cases/wallets';
+import { success } from '@lens-protocol/shared-kernel';
+
+import { PromiseResultPresenter } from '../../transactions/adapters/PromiseResultPresenter';
 
 export type LogoutHandler = (data: LogoutData) => void;
 
-export class LogoutPresenter implements ILogoutPresenter {
-  constructor(private readonly logoutHandler: LogoutHandler) {}
+export class LogoutPresenter
+  extends PromiseResultPresenter<void, never>
+  implements ILogoutPresenter
+{
+  constructor(private readonly logoutHandler: LogoutHandler) {
+    super();
+  }
 
   presentLogout(data: LogoutData): void {
     this.logoutHandler(data);
+    this.present(success());
```

Comment:
Seems convoluted to just reuse a `Deferred`.

The `present` method although defined as part of the public interface at the domain level is only used internally.

It seems that an implementation detail is affecting the domain definition.

I would instead define the `ILogoutPresenter` as:

```ts
type ILogoutPresenter = IGenericResultPresenter<LogoutData, never>
```

then the `LogoutPresenter` can still extend the `PromiseResultPresenter<void, never>`. It can override the `present` method and call the `logoutHandler` before invoking the `super.present(success())`.

Reply:
oh wow, that's a very very good point

Reply:
fixed